lib/src/main/java/uk/gov/di/ipv/cri/passport/library/service/ConfigurationService.java

```diff
@@ -200,4 +203,23 @@ public Certificate getClientJwtSigningCert(String clientId) throws CertificateEx
                                 + "/%s/sharedAttributesJwtSigningCert",
                         clientId));
     }
+
+    public List<String> getClientRedirectUrls(String clientId) {
+        Optional<String> redirectUrlStrings =
+                Optional.ofNullable(
+                        ssmProvider.get(
+                                String.format(
+                                        "/%s/credentialIssuers/ukPassport/clients/%s/jwtAuthentication/validRedirectUrls",
```

Comment:
Just noticed that we have a different pattern for this slightly further up within `getClietnJwtSigningCert` which uses an env var to get the base path of the passport client config.
```
CREDENTIAL_ISSUERS_CONFIG_PARAM_PREFIX
```
Which is set in the SAM template for the JwtVerficationFunction here and would need adding for this function: https://github.com/alphagov/di-ipv-cri-uk-passport-back/blob/8e00f9c25ae98df1de9119a837c7c399633e8c19/deploy/template.yaml#L138

I think that approach was taken because the structure of the params was a bit in flux and using env vars means there's less hard-coding within the lambda app code (although some is still necessary unless we have env vars for every parameter location which might get out of hand...)

A consistent approach might be useful? Something that has got me thinking is that this ConfigurationService is shared across lambdas and although a lambda could therefore use anyone of the functions it exposes, a particular env var that function relies upon may not be set in that lambda's environment, so its shared but might not work without setting up separate config elsewhere (still that's what smoke-test/end-to-end tests are for I guess 🤔 ).

Reply:
Ah yeah good catch. For now I'll change it to use the env var as well to match. Not sure if it could be neater for each lambda to have its own configuration service maybe? But that could end up leading to a fair bit of duplication

Reply:
Yeah I think there's been a bit of discussion around the ConfigurationService because it keeps on growing and lambda are getting access to code that interacts with config they don't need or cannot access anyway. I think the idea behind it being shared is that at its base its doing an often re-usable job of calling SSM for parameters, but perhaps we should split out dealing with a particular parameter or set of parameters into separate classes, so the re-usable ssm interaction is in a common Class and the specific logic for a given param/param path is a little tighter...